### ISSUE TITLE
Cannot authenticate with CLI from VSCode devcontainer

### ISSUE THREAD

**Issue by RiviqVH** [dbt-msft/dbt-sqlserver#192]
I'm having an issue where I can't authenticate to an Azure SQL database using CLI authentication, most simlar to #99 (and somewhat similar to #82). Running on `dbt == 0.21.1` and `dbt-sqlserver == 0.21.1`.

Main difference to above issues is that I am trying to connect from inside a VSCode devcontainer, which is a local Dockerfile based on [this one](https://github.com/microsoft/vscode-dev-containers/tree/main/containers/python-3). On top of this image I add installation for the [MS ODBC 17 Driver for Debian 9](https://docs.microsoft.com/en-us/sql/connect/odbc/linux-mac/installing-the-microsoft-odbc-driver-for-sql-server?view=sql-server-ver15#debian17), [Azure CLI](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli-linux?pivots=apt#option-1-install-with-one-command) and use `pip install` to install `dbt` and `dbt-sqlserver`.

The `devcontainer.json` file points to the image described above. Otherwise, it doesn't add anything significant, just some VSCode extensions and settings.

`profiles.yml` looks as follows:
```
default:
  outputs:
    dev:
      type: sqlserver
      driver: 'ODBC Driver 17 for SQL Server'
      server: <server>
      port: 1433
      database: <database>
      schema: "dbo"
      authentication: CLI

  target: dev
```
Upon opening the devcontainer I run `az login`, which prompts a web page where I can login successfully. Then I run `dbt debug`, which fails with 
```
dbt was unable to connect to the specified database.
The database returned the following error:

  >Database Error
  ('28000', "[28000] [Microsoft][ODBC Driver 17 for SQL Server][SQL Server]Login failed for user ''. (18456) (SQLDriverConnect)")

Check your database credentials and try again. For more information, visit:
https://docs.getdbt.com/docs/configure-your-profile
```

As you see above, it fails with an empty string for user, which is odd (and similar to #99). I made sure that my public IP is whitelisted on the SQL Server, then I retried on my local machine (outside devcontainer), using CLI authentication: connects without a problem. I went back to the devcontainer and tried SQL authentication which works perfectly. 

In short: CLI authentication inside a VSCode devcontainer fails, even though I am logged in using `az login` and even though regular SQL authentication works.

Any ideas how to tackle this further? Thanks in advance!

**Comment by RiviqVH** [comment]
Nevermind, I missed a very obvious thing in Azure: my default subscription was not set correctly. Could've sworn I checked that _before_ posting this issue, but apparently I wasn't thorough enough.
